src/classes/RD_RecurringDonations.cls

```diff
@@ -795,77 +766,391 @@ public class RD_RecurringDonations {
+
+        if (!fixedLengthRDsToRecalculate.isEmpty()) {
+            List<Opportunity> opps = [SELECT Id
+                    , Amount
+                    , Name
+                    , CampaignId
+                    , AccountID
+                    , CloseDate
+                    , StageName
+                    , isWon
+                    , Recurring_Donation_Installment_Number__c
+                    , npe03__Recurring_Donation__c
+            FROM Opportunity
+            WHERE npe03__Recurring_Donation__c IN :fixedLengthRDsToRecalculate
+            ORDER BY CloseDate ASC];
+            handleFixedLengthScheduleRecalculation(fixedLengthRDsToRecalculate, opps, dmlWrapper);
+        }
+    }
+
+    /*******************************************************************************************************
+    * @description Re-evaluates all open Opportunities of an Open-Ended Recurring Donation when it is updated to require schedule recalculation.
+    * @param recurringDonations The list of RDs to process
+    * @param openOpportunities The list of open Opportunities to process
+    * @param dmlWrapper Object to hold the Opps that need deleting or updating
+    * @return void
+    */
+    private static void handleOpenEndedScheduleRecalculation(List<npe03__Recurring_Donation__c> recurringDonations
+            , List<Opportunity> openOpportunities
+            , TDTM_Runnable.DmlWrapper dmlWrapper) {
+
+        npe03__Recurring_Donations_Settings__c rds = UTIL_CustomSettingsFacade.getRecurringDonationsSettings();
+
+        //build a map of recurring donation ID and opps
+        Map<Id, List<Opportunity>> RDOppMap = new Map<Id, List<Opportunity>>();
+        for (Opportunity o : openOpportunities){
+            if (RDOppMap.containsKey(o.npe03__Recurring_Donation__c)) {
+                List<Opportunity> opps = RDOppMap.get(o.npe03__Recurring_Donation__c);
+                opps.add(o);
+                RDOppMap.put(o.npe03__Recurring_Donation__c, opps);
             }
-            else{
-                set<id> rdset = new set<id>();
-                for (npe03__Recurring_Donation__c r : reinsertopps)
-                    rdset.add(r.id);
-                insertOppsOnRecurringDonationInsertFuture(rdset);   
+            else {
+                RDOppMap.put(o.npe03__Recurring_Donation__c, new List<Opportunity>{o});
+            }
+        }
+        // add to the map any RDs that came in with no Opportunities
+        for (npe03__Recurring_Donation__c rd : recurringDonations) {
+            if (!RDOppMap.containsKey(rd.Id)) {
+                RDOppMap.put(rd.Id, new List<Opportunity>());
+            }
+        }
+
+        List<Opportunity> updateOppsList = new List<Opportunity>();
+        List<Opportunity> closeOppsList = new List<Opportunity>();
+        List<Opportunity> insertOppsList = new List<Opportunity>();
+
+        integer forecast_months = (integer)rds.npe03__Opportunity_Forecast_Months__c;
```

Comment:
I believe we've settled on upper case for type declaration ("Integer" instead of "integer"). Plus, the two lines here are not consistent for case - "integer" and "Date"